src/main/java/com/github/technus/tectech/thing/metaTileEntity/multi/GT_MetaTileEntity_EM_EyeOfHarmony.java

```diff
@@ -1131,37 +1193,62 @@ public CheckRecipeResult processRecipe(EyeOfHarmonyRecipe recipeObject) {
                     .insufficientMachineTier((int) recipeObject.getSpacetimeCasingTierRequired());
         }
 
+        // Calculate multipliers used in power calculations
+        double powerMultiplier = Math.max(1, Math.pow(2.3, (parallelExponent - 1)));
+        long precisionMultiplier = (long) Math.pow(10, parallelExponent - 1);
+
+        // Determine EU recipe input
         startEU = recipeObject.getEUStartCost();
 
+        // Determine EU recipe output
+        euOutput = recipeObject.getEUOutput();
+
+        double multiplier = 1;
+
+        // Calculate actual EU values
+        if (parallelAmount > 1) {
+            multiplier = PARALLEL_MULTIPLIER_CONSTANT / POWER_DIVISION_CONSTANT;
+            outputEU_BigInt = BigInteger.valueOf(euOutput)
+                    .multiply(
+                            BigInteger.valueOf(
+                                    (long) (1 - ((TOTAL_CASING_TIERS_WITH_POWER_PENALTY - stabilisationFieldMetadata)
```

Comment:
Shouldn't the output always be positive? It makes more sense if the input value takes all the energy, but the output is `0` until tier `7`. Currently this will extract additional energy when the recipe finishes if the ouput eu is negative.

Reply:
the eu output is never negative, the result from that substraction is always between 0.6 and 1